R/get_admissions.R

```diff
@@ -3,35 +3,60 @@
 #' @description Downloads hospital admissions by Hospital trust using
 #' `download_trust_data` and then optionally aggregates to either LTLA or UTLA
 #' level. This can be done either with the built in mapping or a user supplied mapping.
+#' 
+#' @param keep_vars Character string, defaulting to "new_adm" (first-time COVID-19 hospital admissions). Defines which variables to keep from the raw data. Other supported options are: "all_adm" (for all COVID-19 hospital admissions), and "all_bed" (for all COVID-19 beds occupied). Multiple values allowed.
 #' @param level Character string, defaulting to "trust". Defines the level of aggregation
 #' at which to return the data. Other supported options are "utla" for UTLA level admissions
 #' or "ltla" for LTLA level admissions.
 #' @inheritParams download_trust_data 
 #' @inheritParams get_names
-#' @importFrom dplyr filter select left_join group_by mutate summarise ungroup
-#' @return A data.frame of admissions by day either at trust, LTLA or UTLA levels.
+#' 
+#' @importFrom dplyr filter select left_join group_by mutate summarise pull rename
+#' @importFrom tidyr pivot_wider
+#' @importFrom tibble tibble
+#' 
+#' @return A data.frame of daily admissions and/or bed occupancy data, reported at the Trust, LTLA or UTLA level. Note that new admissions ("new_adm") are called "admissions" in the data.frame to be consistent with a previous version of this function.
 #' @export
-get_admissions <- function(level = "trust", release_date = Sys.Date(), mapping, geo_names) {
+get_admissions <- function(keep_vars = "new_adm",
+                           level = "trust",
+                           release_date = Sys.Date(),
+                           mapping,
+                           geo_names) {
   
-  level <- match.arg(level, choices = c("utla", "ltla", "trust"))
-
+  # Check variables to keep
+  keep_vars <- match.arg(keep_vars, several.ok = TRUE,
+                         choices = c("all_adm", "new_adm", "all_bed"))
```

Comment:
@Bisaloo recently made me aware you can pass a vector of choices to `keep_vars` and the first will be the default. It means you can drop the choices use here if you refactor.